src/sdk/auth/direct-grant-auth-provider.ts

```diff
@@ -1,14 +1,25 @@
 import { AuthProvider, DEFAULT_AUTH_URL, DEFAULT_REALM } from './auth-provider';
 import Axios, { AxiosError, AxiosResponse } from 'axios';
 import querystring = require('querystring');
+import { Subscriber } from 'rxjs/Subscriber';
+import { Observable } from 'rxjs/Observable';
 
 const TOKEN_REFRESH_THRESHOLD = 10;
 
 export class IlandDirectGrantAuthProvider implements AuthProvider {
 
   private _token: Token|undefined;
+  private _tokenObservable: Observable<string>;
 
   constructor(private _config: IlandDirectGrantAuthConfig) {
+    this._tokenObservable = Observable.create((observable: Subscriber<string>) => {
+      this.getToken().then(token => {
+        observable.next(token);
+        observable.complete();
```

Comment:
why are you completing it here? Shouldn't you be continually broadcasting tokens in the refresh method or am I missing something?

Reply:
Well, the `direct-grant-auth-provider` is only used for tests. It will automatically return a connected user/token. There, it doesn't use the keycloak lib at all. It uses a regular Axios request. But because of this, i don't have the event listener to refresh the token.
But, if you think we need to have that to this class, i could create an interval that call the `getToken`method every x seconds. But i don't think it worth it in this particular case.

Reply:
Its not only used for tests, its used for any nodejs client. So yes, we need it to work properly

Reply:
Ok. I'll do the interval solution though. I was thinking of doing one request every seconds. Do you think it makes sense ?

Reply:
there is no need to user an interval.. you can emit from the observer from within the _refresh method since it controls the token refresh exclusively

Reply:
You mean that i just need to emit an event everytime we call the refresh method ?! Yeah i can do it. Better than an interval...

Reply:
Finally i've decided to do like KC (create an `onRefreshToken` method within the class, and assign this method tomy observer and call `onRefreshToken()`from the `_refreshToken()` method. :-)